nuclio_sdk/response.py

```diff
@@ -40,13 +46,68 @@ def ensure_no_ack(self):
         """
         self.headers["x-nuclio-stream-no-ack"] = True
 
+    @staticmethod
+    async def from_entrypoint_output_async(json_encoder, handler_output):
+
+        handler_output_type = Response.get_handler_output_type(handler_output)
+        # Support streaming via sync or async generator
+        if handler_output_type == GENERATOR_RESPONSE:
+            async for chunk in Response.from_generator_output(
+                json_encoder, handler_output
+            ):
+                yield chunk
+
+        # If the handler output is a Response object with a generator body
+        elif handler_output_type == RESPONSE_WITH_GENERATOR_BODY:
+            async for chunk in Response.from_generator_output(
+                json_encoder, handler_output.body
+            ):
+                yield chunk
```

Comment:
Im debating whether merging those two would be more sense, meaning:
```suggestion
        response_output = handler_output
        if handler_output_type == RESPONSE_WITH_GENERATOR_BODY
          response_output = response_output.body
        if handler_output_type in [GENERATOR_RESPONSE, RESPONSE_WITH_GENERATOR_BODY]:
          async for chunk in Response.from_generator_output(
                json_encoder, handler_output
            ):
                yield chunk
        else:
           ...
```

also if `from_generator_output` is async generator then you could return its function all together, no?

Reply:
@liranbg If using return in an async def function without any yield, the function becomes a coroutine, not an async generator and it much less convenient to use it for wrapper.

Reply:
merged first two conditions into one as you suggested ✅